src/QsCompiler/CompilerOptimizations/VariableRenaming.fs

```diff
@@ -12,119 +13,142 @@ open Microsoft.Quantum.QsCompiler.SyntaxTree
+                        return Identifier (LocalVariable (NonNullable<_>.New newName), tArgs)
+                    } |? Identifier (sym, tArgs)
+            }
         }
-    }
 
-    override scope.StatementKind = { new StatementKindTransformation() with
-        override stmtKind.ExpressionTransformation x = scope.Expression.Transform x
-        override stmtKind.LocationTransformation x = scope.onLocation x
-        override stmtKind.ScopeTransformation x = scope.Transform x
-        override stmtKind.TypeTransformation x = scope.Expression.Type.Transform x
-
-        override this.onSymbolTuple syms =
-            match syms with
-            | VariableName item -> VariableName (NonNullable<_>.New (generateUniqueName item.Value))
-            | VariableNameTuple items -> Seq.map this.onSymbolTuple items |> ImmutableArray.CreateRange |> VariableNameTuple
-            | InvalidItem | DiscardedItem -> syms
-
-        override this.onVariableDeclaration stm =
-            let rhs = this.ExpressionTransformation stm.Rhs
-            let lhs = this.onSymbolTuple stm.Lhs
-            QsBinding<TypedExpression>.New stm.Kind (lhs, rhs) |> QsVariableDeclaration
-
-        override this.onForStatement stm =
-            let iterVals = this.ExpressionTransformation stm.IterationValues
-            let loopVar = fst stm.LoopItem |> this.onSymbolTuple
-            let loopVarType = this.TypeTransformation (snd stm.LoopItem)
-            let body = this.ScopeTransformation stm.Body
-            QsForStatement.New ((loopVar, loopVarType), iterVals, body) |> QsForStatement
-
-        override this.onRepeatStatement stm =
-            constants <- enterScope constants
-            skipScope <- true
-            let result = base.onRepeatStatement stm
-            constants <- exitScope constants
-            result
-
-        override this.onQubitScope (stm : QsQubitScope) =
-            let kind = stm.Kind
-            let rhs = this.onQubitInitializer stm.Binding.Rhs
-            let lhs = this.onSymbolTuple stm.Binding.Lhs
-            let body = this.ScopeTransformation stm.Body
-            QsQubitScope.New kind ((lhs, rhs), body) |> QsQubitScope
+        override this.StatementKind = { new StatementKindTransformation() with
+            override __.ExpressionTransformation x = this.Expression.Transform x
+            override __.LocationTransformation x = this.onLocation x
+            override __.ScopeTransformation x = this.Transform x
+            override __.TypeTransformation x = this.Expression.Type.Transform x
+
+            override this.onSymbolTuple syms =
+                match syms with
+                | VariableName item -> VariableName (NonNullable<_>.New (generateUniqueName item.Value))
+                | VariableNameTuple items -> Seq.map this.onSymbolTuple items |> ImmutableArray.CreateRange |> VariableNameTuple
+                | InvalidItem | DiscardedItem -> syms
+
+            override this.onVariableDeclaration stm =
```

Comment:
Please take a look at https://github.com/microsoft/qsharp-compiler/pull/175. 
Merging that in, you should be able to drop the override for onVariableDeclaration, onForStatement, and onQubitScope here.